test/test_pipeline/components/data_preprocessing/test_data_preprocessing.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+import numpy as np
+
+from autosklearn.pipeline.components.data_preprocessing.data_preprocessing \
+    import DataPreprocessor
+
+
+class PreprocessingPipelineTest(unittest.TestCase):
+
+    def do_a_fit_transform(self, sparse_output):
+        # Numerical dataset (as used in NumericalPreprocessingPipelineTest)
+        X_num = np.array([
+            [3.14, 1.,     1.],   # noqa : matrix legibility
+            [3.14, 2., np.nan],   # noqa : matrix legibility
+            [3.14, 3.,     3.]])  # noqa : matrix legibility
+        sdev = (2 / 3) ** .5
+        Y_num = np.array([
+            [-1/sdev, -1/sdev],   # noqa : matrix legibility
+            [     0.,      0.],   # noqa : matrix legibility
+            [ 1/sdev,  1/sdev]])  # noqa : matrix legibility
+        # Categorical dataset (as used in CategoricalPreprocessingPipelineTest)
+        X_cat = np.array([
+            [1, 2, 0],
+            [3, 0, 0],
+            [2, 9, np.nan]])
+        Y_cat = np.array([
+            [1, 0, 0, 0, 1, 0, 0, 1],
+            [0, 0, 1, 1, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0, 1, 1, 0]])
+        # Combined dataset with shuffled columns:
+        X_comb = np.hstack((X_num, X_cat))
+        categ_feat = np.array([False] * 3 + [True] * 3)
+        random_order = np.random.choice(np.arange(6), size=6, replace=False)
+        X_comb = X_comb[:, random_order]
+        categ_feat = categ_feat[random_order]
+        # fit_transform
+        Y_comb = DataPreprocessor(
+            categorical_features=categ_feat, sparse=sparse_output).fit_transform(X_comb)
```

Comment:
Could you please add an additional `transform` here?

Reply:
Yes. I'm just not 100% sure that I understood what you want. So please take a look at the code just to confirm. What I did was to add a transform using the already fitted pipeline but with a different dataset.

Reply:
Yes, that's correct. But please see my other, new comment.